### PR TITLE
Use setuptools-scm to get migration from Git

### PR DESCRIPTION
**Pull request by mgu**
Refs #242

### Successful PR Checklist:

- [ ] Tests
  - [ ] (not applicable?)
- [ ] Documentation
  - [ ] (not applicable?)
- [ ] Had a good time contributing?
- [ ] (Maintainers: add PR labels)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -112,7 +112,7 @@ jobs:
         run: |
           set -xeu
           python -VV
-          pip install setuptools wheel
+          pip install setuptools wheel "setuptools-scm[toml]"
```
Comment (ewjoachim) [inline .github/workflows/main.yml:115]: Is it needed ? I thought the point of putting that in pyproject.toml was that it was THE right place to put it.

Note: this applies to the whole line :)

File: dev-env [hunk 2]
```diff
@@ -22,6 +22,7 @@ echo ""
 
 if ! ls .venv/lib/python*/site-packages/procrastinate.egg-link 1>/dev/null 2>&1; then
     echo "Installing package for development"
+    pip install "setuptools-scm[toml]"
```
Comment (ewjoachim) [inline dev-env:25]: idem ?

File: .github/workflows/main.yml (hunk 1 above)
Comment (mgu) [reply .github/workflows/main.yml:115]: In my tests I had to add it, but I don't know exactly why the build-system was not used when building the sdist/bdist_wheel

**Comment by ewjoachim** [comment]
@mgu If I understand it correctly, there may be a link between pep517-enabled (pyproject.toml) projects and `pip install --editable`.

[This issue is closed](https://github.com/pypa/pip/issues/6314) so I'm not sure what the problem is.

**Comment by ewjoachim** [comment]
After a long and fruitful discussion, we're adding [python-build](https://python-build.readthedocs.io/en/stable/index.html) to the mix.

File: setup.py [hunk 3]
```diff
@@ -1,45 +1,3 @@
...
-        with open(version_filename, "w") as f:
-            f.write(version)
-        yield version
-    finally:
-        os.remove(version_filename)
-
-
-with ensure_version() as version:
-    setuptools.setup(version=version)
+setup()
```
Comment (ewjoachim) [inline setup.py:3]:
```suggestion
setup(use_scm_version=True)
```

**Comment by elemoine** [comment]
FWIW it looks great! :smile:

**Review by elemoine — APPROVED** [review]

**Comment by elemoine** [comment]
Do not forget about the PR labels.

**Review by ewjoachim — APPROVED** [review]
Like my [FedEx driver](https://xkcd.com/1428/) says, move fast & break things
(if things go wrong, we'll iterate)

**Comment by elemoine** [comment]
> Like my [FedEx driver](https://xkcd.com/1428/) says, move fast & break things
> (if things go wrong, we'll iterate)

So long that you're not an aircraft pilot… :-)

**Pull request merged**